sdk/storage/Azure.Storage.DataMovement/src/DataTransferState.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+
+namespace Azure.Storage.DataMovement
+{
+    /// <summary>
+    /// Defines the types of the state a transfer can have.
+    /// </summary>
+    public enum DataTransferState
+    {
+        /// <summary>
+        /// Default value.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// The transfer has been queued up but has not yet started.
+        /// </summary>
+        Queued = 1,
+
+        /// <summary>
+        /// The transfer has started, but has not yet completed.
+        /// </summary>
+        InProgress = 2,
+
+        /// <summary>
+        /// The transfer has started and is in the process of being paused.
+        ///
+        /// Transfer can be stopped if  <see cref="TransferManager.PauseTransferIfRunningAsync(string, System.Threading.CancellationToken)"/>
+        /// or <see cref="DataTransfer.PauseAsync(CancellationToken)"/> is called.
+        /// </summary>
+        Pausing = 3,
+
+        /// <summary>
+        /// The transfer has started and is in the process of being stopped.
+        ///
+        /// Transfer can be stopped if <see cref="DataTransferErrorMode.StopOnAnyFailure"/> is
+        /// enabled in the <see cref="TransferManagerOptions.ErrorHandling"/>.
+        /// </summary>
+        Stopping = 4,
+
+        /// <summary>
+        /// The transfer has been paused. When transfer is paused
+        /// (e.g. see <see cref="TransferManager.PauseTransferIfRunningAsync(string, System.Threading.CancellationToken)"/>)
+        /// during the transfer, this will be the value.
+        /// </summary>
+        Paused = 5,
+
+        /// <summary>
+        /// The transfer has come to a completed state. If the transfer has started and
+        /// has fully stopped will also come to this state.
```

Comment:
I'm not sure of the two things you're trying to distinguish here. Is it "transfer finished" vs "stopped on failure"?

Reply:
I was trying to write out a way to define Completed without making it sound like it was successful transfer without any errors. I think that's why I said that weird sentence of " the transfer has started and has fully stopped" cause I wanted to leave ambiguity that it could have stopped due to an error.